src/PrettierEditProvider.ts

```diff
@@ -86,8 +86,11 @@ class PrettierEditProvider implements
     ): TextEdit[] {
         try {
             return [TextEdit.replace(
-                range,
-                format(document.getText(range), document.fileName)
+                fullDocumentRange(document),
+                format(document.getText(), document.fileName, {
+                  rangeStart: document.offsetAt(range.start),
+                  rangeEnd: document.offsetAt(range.end)
```

Comment:
Yup, and there's also [`positionAt`](https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextDocument.positionAt) for going the other way.